test/e2e/network/loadbalancer.go

```diff
@@ -54,6 +54,42 @@ import (
 	"github.com/onsi/gomega"
 )
 
+// getSubnetPrefix gets first 2 number of an IP in the node subnet. [IPv4]
+// It assumes that the subnet mask is /16.
+func getSubnetPrefix(c clientset.Interface) ([]string, error) {
+	node, err := getReadySchedulableWorkerNode(c)
+	if err != nil {
+		return nil, fmt.Errorf("error getting a ready schedulable worker Node, err: %v", err)
+	}
+	internalIP, err := e2enode.GetInternalIP(node)
+	if err != nil {
+		return nil, fmt.Errorf("error getting Node internal IP, err: %v", err)
+	}
+	splitted := strings.Split(internalIP, ".")
+	if len(splitted) == 4 {
+		return splitted[:2], nil
+	}
+	return nil, fmt.Errorf("invalid IP address format: %s", internalIP)
+}
```

Comment:
can't comment on untouched code, after returning a net.IPnet you can use net.IPNet.Contains and replace the `strings.HasPrefix(ingressEndpoint, subnetPrefix[0]+".")` for using IPNet and IP comparisons

Reply:
https://pkg.go.dev/net#IPNet.Contains

Reply:
👍  Thank you for the detailed guidance, will do these changes in follow up commits (following suggestions to do multiple commits for easier review)

Reply:
take your time, and check the code around, maybe you can come up with some better ideas